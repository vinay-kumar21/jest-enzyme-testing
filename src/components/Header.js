import React from 'react';
import './styles.scss';
import Logo from '../assets/logo.png'
function Header() {
  return (
    <header >
      <div className="wrap">
                <div className="logo">
                    <img src={Logo} alt="Logo" />
              
                </div>
            </div>
    </header>
  );
}

export default Header;
