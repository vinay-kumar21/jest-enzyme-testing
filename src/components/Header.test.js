import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
import { findByTestAtrr} from '../Utils/Utils';

const setUp = (props = {}) => {
  const component = shallow(<Header {...props} />);
  return component;
};

describe("Header Component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("renders without any error", () => {
    const wrapper = findByTestAtrr(component, "headerComponent");

    expect(wrapper.length).toBe(1);
  });

  it("Should render a logo", () => {
    const logoImg = findByTestAtrr(component, "logoIMG");

    expect(logoImg.length).toBe(1);
  });

  it('Snapshot test of Header Component',()=>{
      const snapShot=findByTestAtrr(component,'headerComponent');
      expect(snapShot).toMatchSnapshot();
  })
});
