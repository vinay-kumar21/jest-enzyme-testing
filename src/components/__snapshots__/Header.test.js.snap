// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot test of Header Component 1`] = `ShallowWrapper {}`;
